lib/cloudregister/registerutils.py

```diff
@@ -1407,6 +1392,26 @@ def write_framework_identifier(cfg):
         framework_file.write(json.dumps(identifier))
 
 
+# ----------------------------------------------------------------------------
+def can_make_ip_request(rmt_ip):
+    """Check if the instance can make IPv4 or IPv6 requests."""
+    try:
+        __connection_check(rmt_ip, True)
```

Comment:
We do not need to make a connection check to determine if the client has IPv6 or IPv4 internet capability. Also as stated the connection check is basically the same as `is_responsive`

Reply:
the method `is_responsive` checks that any RMT update server (either IPv6 or IPv4) are reachable,
we can not know if the instance has IPv6 enabled only with `is_responsive`  as the list of IPs checked on
`is_responsive` could be _only_ IPv4, it could return `True` but that would not confirm IPv6 enabled on
the client, so that method does not guarantee enablement of IPv4/IPv6 on the client 
Another example, if RMT provided IPv6 addresses and the client did not have IPv6 enabled, it would generate
an exception, pass because that's how the method is implemented and would try IPv4 after the IPv6 attempts,
let's say IPv4 succeeded, `is_responsive` would return `True` and we would still not know if the client has
IPv6 enabled
I understand the intend of `is_responsive` what I mean is it is benefitial to tell the user that they are trying 
something that they can't/do not have enabled, `is_responsive` can't tell that
I understand this PR/check is for `smt-ip` in the `force-new` path, but the [IPv4|IPv6] check should be in place
for other parts of this codebase, like [this check](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L268) that checks is always `True` even when the instance 
does not have IPv6 enabled 

IMO, we should have a way to check the instance/client has IPv4 or IPv6 enabled, whether using `is_responsive` 
or not I dont really mind, currently `is_responsive` does not guarantee whether the client has IPv4/IPv6 enabled or not

Reply:
> 
> IMO, we should have a way to check the instance/client has IPv4 or IPv6 enabled, 

I am agree wit this idea. But implementing the concepts of `is_responsive` is not the way to achieve that goal.

That said when the user specifies update server information then we end up here https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/usr/sbin/registercloudguest#L298 where I can insert:

```
smt = smt.SMT(region_smt_data)
if not smt.is_responsive():
    print('Provided update server IP address %s with "--smt-ip" is not reachable' % args.user_smt_ip')
    sys.exit(1)
```

So what you state in the description as a goal to reduce debugging time and provide an early exit condition can be achieved in 4 lines of code with any refactoring needed and we tell the user that the information provided cannot be used, whether it is IPv4 or IPv6 is immaterial.

So it appears that the mixing of "it would be nice to know if the instance has IPv4, IPV6 or both" and "let's have an early exit condition when the user gives us a bad ip-address" is creating problems with the clarity of implementation.